client/my-sites/checkout/checkout-thank-you/test/index.js

```diff
@@ -85,31 +106,43 @@ describe( 'CheckoutThankYou', () => {
 			refreshSitePlans: ( selectedSite ) => selectedSite,
 			planSlug: PLAN_BUSINESS,
 		};
+
 		test( 'Should display a full version when isSimplified is missing', () => {
-			const comp = shallow( <CheckoutThankYou { ...props } /> );
-			expect( comp.find( '.checkout-thank-you__purchase-details-list' ) ).toHaveLength( 1 );
-			expect( comp.find( 'HappinessSupport' ) ).toHaveLength( 1 );
-			expect( comp.find( 'CheckoutThankYouHeader' ).props().isSimplified ).toBeFalsy();
+			render( <CheckoutThankYou { ...props } /> );
+			expect( screen.queryByTestId( 'business-plan-details' ) ).toBeVisible();
+			expect( screen.queryByTestId( 'happiness-support' ) ).toBeVisible();
+			expect( CheckoutThankYouHeader ).toHaveBeenCalledWith(
```

Comment:
I generally recommend mocking `CheckoutThankYouHeader` and rendering the props we're interested in right there in the mock, instead of relying on the API with which React renders components. WDYT?

Reply:
It is mocked (see line 35). I think `jest.fn` and `.toHaveBeenCalledWith` provide a good way to test for passed on props.

Reply:
It definitely does the job, so I'm leaving this to your personal preference.

Reply:
In general though I don't think testing for passed on props is a solid enough way to test the component itself and rather a legacy from previous testing habits. Passing on props on its own doesn't say a lot to the user so we shouldn't rely only on them but make sure there's tests which test the actual result of those passed on props (difference in rendering, etc).

Reply:
Agreed, but testing some other component's rendered result steps into its own tests' territory, which is why testing passed props is generally enough. What the component actually does with the props is a responsibility of the tests of that component.

Reply:
Agreed, too. I wonder whether this would be a good reason to use an inline snapshot with those rendered props (by rendering those as you suggested) to make sure there's no unintended changes instead of manually asserting those 🤔

Reply:
Snapshots can and will include way more than we're interested in here (a single prop), so I think it's not a proper replacement. Any change to a component in the component tree (completely unrelated to the one we care about) will trigger a failure, and that'd be undesired IMHO.

Reply:
Not a replacement, true. I was thinking more about the reasoning behind testing passed on props without testing the actual result. What came to mind is we want to make sure it doesn't get removed accidentally (i.e. unintended changes) and how that compares to other props we pass on.

Reply:
Gotcha. Well, if we mock the extra components (including that one that receives the props) that could save us from extra unrelated changes breaking the snapshots. Is that what you have in mind?